src/text/non_blocking/raw_text_reader.rs

```diff
@@ -335,6 +334,12 @@ impl<T: ToIonDataSource> RawTextReader<T> {
                             // Retry the parser on the extended buffer in the next loop iteration
                             continue;
                         }
+                        Err(TextError::Incomplete { line, column }) => {
+                            // If load_next_line() returns TextError::Incomplete, then that means
+                            // it has incomplete UTF-8 data in the buffer, suggesting the user is
+                            // not done providing the ion data.
+                            return RootParseResult::Incomplete(line, column);
+                        }
```

Comment:
For the purposes of this blocking reader, should this just `continue`? There's incomplete data at the end of the line (but maybe also new bytes before that) so just try parsing again?

Reply:
> My understanding is that all multi-byte sequences should be in some sort of container (a string, clob, ''d symbol, etc), is that right?

Yep!